pkg/splunk/enterprise/finalizers_test.go

```diff
@@ -149,7 +149,7 @@ func splunkDeletionTester(t *testing.T, cr splcommon.MetaObject, delete func(spl
 				{MetaName: "*v1.StatefulSet-test-splunk-test-monitoring-console"},
 				{MetaName: "*v1.Secret-test-splunk-test-secret"},
 				{MetaName: "*v1.Secret-test-splunk-test-secret"},
-				{MetaName: "*v1.ClusterMaster-test-master1"},
+				{MetaName: "*v2.ClusterMaster-test-master1"},
```

Comment:
This is interesting.. why do we need to change the version of ClusterMaster-test-master1 alone?

Reply:
This is happening, when the IndexerCluster CR is configured with `ClusterMasterRef: and when the ApplyIndexerCluster() is called. In my first set of changes with the soft link mechanism, thought of addressing it by dynamically passing the latest version as below, but for now kept it aside, as it may not worth just addressing a UT case. We can fix it later when we address it for the main code:

```
func getTheLatestCRDVersion() (string, error) {
	// Get the actual dir entry name pointed by the softlink `latest` relative to the path ../splunk-operator/pkg/splunk/enterprise/
	return os.Readlink("../../apis/enterprise/latest")
}
func TestApplyIndexerCluster(t *testing.T) {
	latestCrdVersion, err := getTheLatestCRDVersion()
	if err != nil {
		t.Errorf("Unable to find the latest CRD version. Error: %v", err)
	}

	mockFuncCallForCmWithLatestCrdVersion := "*" + latestCrdVersion + ".ClusterMaster-test-master1"

	funcCalls := []spltest.MockFuncCall{
		{MetaName: "*v1.Secret-test-splunk-test-secret"},
		{MetaName: "*v1.Secret-test-splunk-test-secret"},
		{MetaName: mockFuncCallForCmWithLatestCrdVersion},
		{MetaName: "*v1.Service-test-splunk-stack1-indexer-headless"},
		{MetaName: "*v1.Service-test-splunk-stack1-indexer-service"},
		{MetaName: "*v1.Secret-test-splunk-test-secret"},
		{MetaName: "*v1.Secret-test-splunk-stack1-indexer-secret-v1"},
		{MetaName: mockFuncCallForCmWithLatestCrdVersion},
		{MetaName: "*v1.Secret-test-splunk-test-secret"},
	}

Reply:
Follow up question: Why is the version change only for ClusterMaster-test-master1? Is it because it is an enterprise splunk API?

Reply:
Ya, as it is our API.